File: SwiftOSXProofOfConcept/data2yaml/data/jitter.lua
sptbl["jitter"] = {

    files = { 
        module = "jitter.c",
        header = "jitter.h",
        example = "ex_jitter.c",
    },
    
    func = {
        create = "sp_jitter_create",
        destroy = "sp_jitter_destroy",
        init = "sp_jitter_init",
        compute = "sp_jitter_compute",
    },
    
    params = {
        optional = {
            {
                name = "amp",
                type = "SPFLOAT",
                description ="The amplitude of the line. Will produce values in the range of (+/-)amp.",
                default = 0.5
            },
            {
                name = "cpsMin",
                type = "SPFLOAT",
                description = "The minimum frequency of change in Hz.",
                default = 0.5
            },
            {
                name = "cpsMax",
                type = "SPFLOAT",
                description ="The maximum frequency of change in Hz.",
                default = 4
            },
        }
    },
    
    modtype = "module",
    
    description = [[Produce a signal with random fluctuations (aka... jitter). This is useful for emulating jitter found in analogue equipment. ]], 
    
    ninputs = 0,
    noutputs = 1,
    
    inputs = { 
        {
            name = "dummy",
            description = "This doesn't do anything."
        },
    },
    
    outputs = {
        {
            name = "out",
            description = "Signal output."
        },
    }

}

File: Sources/AudioKit/Operations/Math/Random Number Generators/jitter.swift
// Copyright AudioKit. All Rights Reserved. Revision History at http://github.com/AudioKit/AudioKit/

extension Operation {

    /// A signal with random fluctuations
    /// This is useful for emulating jitter found in analogue equipment.
    ///
    /// - Parameters:
    ///   - amplitude: The amplitude of the line. Will produce values in the range of (+/-)amp. (Default: 0.5)
    ///   - minimumFrequency: The minimum frequency of change in Hz. (Default: 0.5)
    ///   - maximumFrequency: The maximum frequency of change in Hz. (Default: 4)
    ///
    public static func jitter(
        amplitude: OperationParameter = 0.5,
        minimumFrequency: OperationParameter = 0.5,
        maximumFrequency: OperationParameter = 4
        ) -> Operation {
        return Operation(module: "jitter",
                           inputs: amplitude, minimumFrequency, maximumFrequency)
    }
}
